README.md

```diff
@@ -4,9 +4,43 @@ This is a sample application for integrating libaktualizr. The application is in
 
 To build and run the application locally:
 ```bash
-# mkdir build && cd build
-# cmake ..
-# make libaktualizr-demo-app
-# cp /path/to/credentials.zip ./credentials.zip
-# src/libaktualizr-demo-app -c ../aktualizr/config/sota-local.toml
+mkdir build && cd build
+cmake ..
+make libaktualizr-demo-app
+cp /path/to/credentials.zip ./credentials.zip
+src/libaktualizr-demo-app -c ../aktualizr/config/sota-local.toml
 ```
+
+## What does this do?
+
+libaktualizr provides API calls to perform the various steps that are necessary for checking for updates, validating the files downloaded, and so on. This demo app is a simple command line app that acts as a kind of REPL for aktualizr. It takes a normal aktualizr config file and credentials file, registers/provisions with the server if it's not already been done, and then waits for commands.
+
+The commands it can accept are all of the commands from the `aktualizr` class that are needed to check for, download, and install updates and campaigns:
+
+* SendDeviceData
+* CheckUpdates
+* Download
+* Install
+* CampaignCheck
+* CampaignAccept
+* Pause
+* Resume
+* Abort
+
+You can see a description of what each command does in the [aktualizr class reference in libaktualizr's Doxygen docs.](https://advancedtelematic.github.io/aktualizr/class_aktualizr.html).
```

Comment:
If we're gonna link straight to the DOxygen, then we need to up our game with regards to function descriptions:
See the Google Guidelines for function descriptions: https://google.github.io/styleguide/cppguide.html#Function_Comments

>Types of things to mention in comments at the function declaration:
>
> What the inputs and outputs are.
> For class member functions: whether the object remembers reference arguments beyond the duration of the method call, and whether it will free them or not.
> If the function allocates memory that the caller must free.
> Whether any of the arguments can be a null pointer.
> If there are any performance implications of how a function is used.
> If the function is re-entrant. What are its synchronization assumptions?

The Uptane related functions/terminology need extra attention....ie: what's an "Uptane Cycle"?
* Do we expected Lucid or UNU devs to understand what that means?

Reply:
"Uptane Cycle" isn't used or referenced in the demo app.

Taking the google list, with all due respect, I can't see anything relevant that isn't there already. Can you be more specific about which of those things you think are missing, and where?

Reply:
Well I was just expecting a little more detail in the descriptions, something along the the lines of the [Qt ref doc](https://doc.qt.io/qt-5/qsavefile.html#commit). I'll get back to you with more specifics but since this is about the Doxygen rather than this specific change, it's not a blocker for this PR.